staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/grpc_service_unix_test.go

```diff
@@ -153,18 +156,28 @@ func TestTimeouts(t *testing.T) {
 
 				f, err := mock.NewBase64Plugin(socketName.path)
 				if err != nil {
-					t.Fatalf("failed to construct test KMS provider server, error: %v", err)
+					kmsPluginErrc <- fmt.Errorf("failed to construct test KMS provider server, error: %v", err)
+					return
 				}
 				if err := f.Start(); err != nil {
-					t.Fatalf("Failed to start test KMS provider server, error: %v", err)
+					kmsPluginErrc <- fmt.Errorf("Failed to start test KMS provider server, error: %v", err)
+					return
 				}
 				defer f.CleanUp()
 				kmsPluginWG.Done()
 				// Keeping plugin up to process requests.
 				testCompletedWG.Wait()
 			}()
 
-			kubeAPIServerWG.Wait()
+			go func() {
+				kubeAPIServerWG.Wait()
+				close(kubeAPIServerErrc)
+			}()
```

Comment:
why do we need to run 173-174 in a goroutine?

Reply:
`kubeAPIServerWG.Done()` in [L147](https://github.com/kubernetes/kubernetes/pull/103023/files/9bcdc8e06e3a7a9bcbcbfe8b74596af6ffa3e4f9#diff-8e0703b57f74e30387190cfa425dacd197fced93b315b316b4e18639e6756f6aR147) is only being called if `NewGRPCService(...)` in [L141](https://github.com/kubernetes/kubernetes/pull/103023/files/9bcdc8e06e3a7a9bcbcbfe8b74596af6ffa3e4f9#diff-8e0703b57f74e30387190cfa425dacd197fced93b315b316b4e18639e6756f6aR141-R144)  doesn't return an error. It might block, and a potential error wouldn't be reported otherwise.

Reply:
I see. Then if L141 does return an error, then this goroutine is leaked, right?

Can we make L141 call `kubeAPIServerWG.Done()` in both err and non-err cases, and move L173-L174 to run in the main goroutine?

Reply:
You're right, applied your feedback in 987fd88.